cmd/image-loader/main.go

```diff
@@ -65,34 +68,42 @@ var (
 )
 
 type opts struct {
-	versionsFile  string
-	versionFilter string
-	registry      string
-	dryRun        bool
-	addonsPath    string
+	configurationFile string
+	versionsFile      string
+	versionFilter     string
+	registry          string
+	dryRun            bool
+	addonsPath        string
 }
 
 func main() {
-	klog.InitFlags(nil)
+	var err error
 
 	logOpts := kubermaticlog.NewDefaultOptions()
+	logOpts.Format = kubermaticlog.FormatConsole
 	logOpts.AddFlags(flag.CommandLine)
 
 	o := opts{}
-	flag.StringVar(&o.versionsFile, "versions", "charts/kubermatic/static/master/versions.yaml", "The versions.yaml file path")
+	flag.StringVar(&o.configurationFile, "configuration-file", "", "Path to the KubermaticConfiguration YAML file")
+	flag.StringVar(&o.versionsFile, "versions-file", "", "The versions.yaml file path (deprecated, EE-only, used only if no -configuration-file is given)")
 	flag.StringVar(&o.versionFilter, "version-filter", "", "Version constraint which can be used to filter for specific versions")
-	flag.StringVar(&o.registry, "registry", "registry.corp.local", "Address of the registry to push to")
+	flag.StringVar(&o.registry, "registry", "", "Address of the registry to push to, for example localhost:5000")
 	flag.BoolVar(&o.dryRun, "dry-run", false, "Only print the names of found images")
 	flag.StringVar(&o.addonsPath, "addons-path", "", "Path to the folder containing the addons")
 	flag.Parse()
 
-	log := kubermaticlog.New(logOpts.Debug, logOpts.Format)
+	rawLog := kubermaticlog.New(logOpts.Debug, logOpts.Format)
+	log := rawLog.Sugar()
 	defer func() {
 		if err := log.Sync(); err != nil {
 			fmt.Println(err)
```

Comment:
Locally (macos) I'm getting this output when the process terminates:
```
sync /dev/stderr: inappropriate ioctl for device
```
I think we can safely avoid logging `Sync` errors and do as mentioned in the quickstart: https://github.com/uber-go/zap#quick-start

```
rawLog := kubermaticlog.New(logOpts.Debug, logOpts.Format)
defer rawLog.Sync()
log := rawLog.Sugar()
```

Reply:
This should then affect all binaries we have, no? This is not the PR to touch most of our binaries ;-)

Reply:
If we do the same probably affects the other binaries too. But this seems to be OS specific and only CLIs are supposed to run on macos.

Reply:
Anyway, if you prefer to handle this separately it's not a problem as this issues was not generated by your changes.